src/System.Net.HttpListener/src/System/Net/Managed/HttpListenerResponse.Managed.cs

```diff
@@ -408,7 +405,10 @@ internal void SendHeaders(bool closing, MemoryStream ms, bool isWebSocketHandsha
 
             /* Assumes that the ms was at position 0 */
             ms.Position = preamble;
-            _headersSent = !isWebSocketHandshake;
+            if (!isWebSocketHandshake)
+            {
+                _responseState = ResponseState.SentHeaders;
+            }
```

Comment:
If it is a web socket handshake, previously _headersSet would have been set to false.  Does _responseState need to be set to something in that case, or it's ok leaving whatever its previous value was?